### PR TITLE
Allow no schema stamping during update

### LINKED ISSUE #1255: Force no stamping on schema upgrade
Stamping during shema upgrade does not work reliably in the tool "privacyidea-schema-upgrade".
We should add a parameter to "force no stamp" and also allow this parameter in "privacyidea-pip-update".

### PR DESCRIPTION
**Pull request by cornelinux**
The parameter -s can be used to skip the
schema stamping during the update process.
THis can help in certain installations,
where the version of the schema can not be
automatically determined.

Closes #1255
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23issuecomment-425424352%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23discussion_r221531742%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23discussion_r221531837%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23discussion_r221532025%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23discussion_r221534189%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23discussion_r221534264%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23discussion_r221534286%22%2C%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23issuecomment-425851449%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23issuecomment-425424352%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%23%20%5BCodecov%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1265%3Fsrc%3Dpr%26el%3Dh1%29%20Report%5Cn%3E%20Merging%20%5B%231265%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1265%3Fsrc%3Dpr%26el%3Ddesc%29%20into%20%5Bmaster%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/commit/6f16fab048c6aa7f6209cba872492213ad652cdd%3Fsrc%3Dpr%26el%3Ddesc%29%20will%20%2A%2Anot%20change%2A%2A%20coverage.%5Cn%3E%20The%20diff%20coverage%20is%20%60n/a%60.%5Cn%5Cn%5B%21%5BImpacted%20file%20tree%20graph%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1265/graphs/tree.svg%3Fwidth%3D650%26token%3D7nHLZki60B%26height%3D150%26src%3Dpr%29%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1265%3Fsrc%3Dpr%26el%3Dtree%29%5Cn%5Cn%60%60%60diff%5Cn%40%40%20%20%20%20%20%20%20%20%20%20%20Coverage%20Diff%20%20%20%20%20%20%20%20%20%20%20%40%40%5Cn%23%23%20%20%20%20%20%20%20%20%20%20%20master%20%20%20%20%231265%20%20%20%2B/-%20%20%20%23%23%5Cn%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%5Cn%20%20Coverage%20%20%2095.82%25%20%20%2095.82%25%20%20%20%20%20%20%20%20%20%20%20%5Cn%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%5Cn%20%20Files%20%20%20%20%20%20%20%20%20141%20%20%20%20%20%20141%20%20%20%20%20%20%20%20%20%20%20%5Cn%20%20Lines%20%20%20%20%20%20%2017150%20%20%20%2017150%20%20%20%20%20%20%20%20%20%20%20%5Cn%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%5Cn%20%20Hits%20%20%20%20%20%20%20%2016434%20%20%20%2016434%20%20%20%20%20%20%20%20%20%20%20%5Cn%20%20Misses%20%20%20%20%20%20%20%20716%20%20%20%20%20%20716%5Cn%60%60%60%5Cn%5Cn%5Cn%5Cn------%5Cn%5Cn%5BContinue%20to%20review%20full%20report%20at%20Codecov%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1265%3Fsrc%3Dpr%26el%3Dcontinue%29.%5Cn%3E%20%2A%2ALegend%2A%2A%20-%20%5BClick%20here%20to%20learn%20more%5D%28https%3A//docs.codecov.io/docs/codecov-delta%29%5Cn%3E%20%60%5Cu0394%20%3D%20absolute%20%3Crelative%3E%20%28impact%29%60%2C%20%60%5Cu00f8%20%3D%20not%20affected%60%2C%20%60%3F%20%3D%20missing%20data%60%5Cn%3E%20Powered%20by%20%5BCodecov%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1265%3Fsrc%3Dpr%26el%3Dfooter%29.%20Last%20update%20%5B6f16fab...df524ea%5D%28https%3A//codecov.io/gh/privacyidea/privacyidea/pull/1265%3Fsrc%3Dpr%26el%3Dlastupdated%29.%20Read%20the%20%5Bcomment%20docs%5D%28https%3A//docs.codecov.io/docs/pull-request-comments%29.%5Cn%22%2C%20%22created_at%22%3A%20%222018-09-28T12%3A46%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/in/254%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/apps/codecov%22%7D%7D%2C%20%7B%22body%22%3A%20%22Looks%20good.%22%2C%20%22created_at%22%3A%20%222018-10-01T09%3A54%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/37443810%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/plettich%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20df524ea63da246a1905eb17a7c1cf25cb0ce9547%20doc/installation/upgrade.rst%2010%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23discussion_r221532025%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22...%20and%20...%3F%22%2C%20%22created_at%22%3A%20%222018-10-01T08%3A44%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/37443810%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/plettich%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%20%3A-%29%22%2C%20%22created_at%22%3A%20%222018-10-01T08%3A52%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/1908620%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cornelinux%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20doc/installation/upgrade.rst%3AL37-51%22%7D%2C%20%22Pull%20df524ea63da246a1905eb17a7c1cf25cb0ce9547%20doc/installation/upgrade.rst%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23discussion_r221531742%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22...%20allowed%3A%22%2C%20%22created_at%22%3A%20%222018-10-01T08%3A43%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/37443810%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/plettich%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222018-10-01T08%3A51%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/1908620%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cornelinux%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20doc/installation/upgrade.rst%3AL37-51%22%7D%2C%20%22Pull%20df524ea63da246a1905eb17a7c1cf25cb0ce9547%20doc/installation/upgrade.rst%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/privacyidea/pull/1265%23discussion_r221531837%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22...%20want%20...%22%2C%20%22created_at%22%3A%20%222018-10-01T08%3A44%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/37443810%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/plettich%22%7D%7D%2C%20%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222018-10-01T08%3A52%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/1908620%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cornelinux%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20doc/installation/upgrade.rst%3AL37-51%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/privacyidea/privacyidea/pull/1265#issuecomment-425424352'>General Comment</a></b>
- <a href='https://github.com/apps/codecov'></a> # [Codecov](https://codecov.io/gh/privacyidea/privacyidea/pull/1265?src=pr&el=h1) Report
> Merging [#1265](https://codecov.io/gh/privacyidea/privacyidea/pull/1265?src=pr&el=desc) into [master](https://codecov.io/gh/privacyidea/privacyidea/commit/6f16fab048c6aa7f6209cba872492213ad652cdd?src=pr&el=desc) will **not change** coverage.
> The diff coverage is `n/a`.
[Impacted file tree graph](https://codecov.io/gh/privacyidea/privacyidea/pull/1265?src=pr&el=tree)
```diff
@@           Coverage Diff           @@
##           master    #1265   +/-   ##
=======================================
Coverage   95.82%   95.82%
=======================================
Files         141      141
Lines       17150    17150
=======================================
Hits        16434    16434
Misses        716      716
```
------
[Continue to review full report at Codecov](https://codecov.io/gh/privacyidea/privacyidea/pull/1265?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/privacyidea/privacyidea/pull/1265?src=pr&el=footer). Last update [6f16fab...df524ea](https://codecov.io/gh/privacyidea/privacyidea/pull/1265?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).
- <a href='https://github.com/plettich'></a> Looks good.
- [ ] <a href='#crh-comment-Pull df524ea63da246a1905eb17a7c1cf25cb0ce9547 doc/installation/upgrade.rst 4'></a> &nbsp;<b><a href='https://github.com/privacyidea/privacyidea/pull/1265#discussion_r221531742'>File: doc/installation/upgrade.rst:L37-51</a></b>
- <a href='https://github.com/plettich'></a> ... allowed:
- <a href='https://github.com/cornelinux'></a> done
- [ ] <a href='#crh-comment-Pull df524ea63da246a1905eb17a7c1cf25cb0ce9547 doc/installation/upgrade.rst 6'></a> &nbsp;<b><a href='https://github.com/privacyidea/privacyidea/pull/1265#discussion_r221531837'>File: doc/installation/upgrade.rst:L37-51</a></b>
- <a href='https://github.com/plettich'></a> ... want ...
- <a href='https://github.com/cornelinux'></a> done
- [ ] <a href='#crh-comment-Pull df524ea63da246a1905eb17a7c1cf25cb0ce9547 doc/installation/upgrade.rst 10'></a> &nbsp;<b><a href='https://github.com/privacyidea/privacyidea/pull/1265#discussion_r221532025'>File: doc/installation/upgrade.rst:L37-51</a></b>
- <a href='https://github.com/plettich'></a> ... and ...?
- <a href='https://github.com/cornelinux'></a> done :-)

<a href='https://www.codereviewhub.com/privacyidea/privacyidea/pull/1265?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/privacyidea/privacyidea/pull/1265?approve=1'></a>&nbsp;<a href='https://github.com/privacyidea/privacyidea/pull/1265'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: doc/installation/upgrade.rst [hunk 1]
```diff
@@ -37,6 +37,14 @@ backup) and run the command:
 
    privacyidea-pip-update
 
+The following parameters are allows:
```
Comment (plettich) [inline doc/installation/upgrade.rst]: ... allowed:

File: doc/installation/upgrade.rst [hunk 2]
```diff
@@ -37,6 +37,14 @@ backup) and run the command:
...
+
+``-f`` or ``--force`` skips the safety question, if you really wnat to update.
```
Comment (plettich) [inline doc/installation/upgrade.rst]: ... want ...

File: doc/installation/upgrade.rst [hunk 3]
```diff
@@ -37,6 +37,14 @@ backup) and run the command:
...
+
+``-s`` or ``--skipstamp`` skips the version stamping during schema update.
+
+``-n`` or ``--noshema`` completely skips the schema update an only updates the code.
```
Comment (plettich) [inline doc/installation/upgrade.rst]: ... and ...?

File: doc/installation/upgrade.rst (hunk 1 above)
Comment (cornelinux) [reply doc/installation/upgrade.rst]: done

File: doc/installation/upgrade.rst (hunk 2 above)
Comment (cornelinux) [reply doc/installation/upgrade.rst]: done

File: doc/installation/upgrade.rst (hunk 3 above)
Comment (cornelinux) [reply doc/installation/upgrade.rst]: done :-)

**Comment by plettich** [comment]
Looks good.

**Pull request merged**